### ISSUE TITLE
Prompt switch network if a user's wallet is connected to the wrong network

### ISSUE THREAD

**Issue by willyogo** [ATXDAO/nouns-monorepo#54]
As a user, I want the member portal/auction app to prompt my wallet to switch if I'm not connected to the right network and attempt to connect wallet or prompt a transaction.

AC:
1. If a user is not on either ethereum mainnet (prod) or goerli (staging), and attempts to connects a wallet, the app should prompt the user's wallet to switch network
2. Whenever a user is prompted to execute a transaction (ie. place bid or settle auction), the app first checks to see which network they're connected to
3. If a user rejects the request to switch network, their wallet is not connected or prompted to execute the transaction (iow, nothing happens and the user can attempt to connect wallet or execute transaction again. if they have switched to the correct network it will work, if not they should get prompted to switch network again. this is the existing behavior for the Switch button on the Rep page)

<img width="522" alt="image" src="https://github.com/ATXDAO/nouns-monorepo/assets/26232238/5763fb26-9c86-4b91-8d42-edc921ef71d5">

here's an article on how to do this: https://www.freecodecamp.org/news/how-to-switch-blockchains-on-metamask-with-javascript/

we also have this functionality already rocking on the Rep page

<img width="1289" alt="image" src="https://github.com/ATXDAO/nouns-monorepo/assets/26232238/edfef1dc-bdf5-4721-a5eb-df75eec05db6">

**Comment by JacobHomanics** [comment]
@willyogo Should we remove AC 1 as that popup prevents them from continuing to use the app even if they have no intention of making any transactions? Then only when they attempt to make a transaction is when we check and prompt for a network switch. 

Going to to hold off on AC 2 as there were changes to the appropriate code in [PR 56](https://github.com/ATXDAO/nouns-monorepo/pull/56). Thus once that PR gets merged in, then we can begin work on AC 2.

AC 3 is also has the same contingencies as AC 2.

**Comment by willyogo** [comment]
> @willyogo Should we remove AC 1 as that popup prevents them from continuing to use the app even if they have no intention of making any transactions? Then only when they attempt to make a transaction is when we check and prompt for a network switch.
> 
> Going to to hold off on AC 2 as there were changes to the appropriate code in [PR 56](https://github.com/ATXDAO/nouns-monorepo/pull/56). Thus once that PR gets merged in, then we can begin work on AC 2.
> 
> AC 3 is also has the same contingencies as AC 2.

great point that users don't really need to switch network unless they're actually going to execute a transaction, which is awesome. is this true for the rep page too? if so, let's just remove that pop up so users aren't prevented from accessing the dapp if they're on the wrong network, and then AC 2 and 3 will take care of ensuring they're on the right network before executing a tx (it's actually possible to get into a state currently where the auction page will let you execute a bid or settle transaction on the wrong network, which can lead to loss of bid 😬).

ty ser!

**Comment by JacobHomanics** [comment]
> > @willyogo Should we remove AC 1 as that popup prevents them from continuing to use the app even if they have no intention of making any transactions? Then only when they attempt to make a transaction is when we check and prompt for a network switch.
> > Going to to hold off on AC 2 as there were changes to the appropriate code in [PR 56](https://github.com/ATXDAO/nouns-monorepo/pull/56). Thus once that PR gets merged in, then we can begin work on AC 2.
> > AC 3 is also has the same contingencies as AC 2.
> 
> great point that users don't really need to switch network unless they're actually going to execute a transaction, which is awesome. is this true for the rep page too? if so, let's just remove that pop up so users aren't prevented from accessing the dapp if they're on the wrong network, and then AC 2 and 3 will take care of ensuring they're on the right network before executing a tx (it's actually possible to get into a state currently where the auction page will let you execute a bid or settle transaction on the wrong network, which can lead to loss of bid 😬).
> 
> ty ser!

Yup! Just updated the PR with changes to properly switch network if you try to execute a tx on the wrong network!

**Comment by willyogo** [comment]
nice! do you know which PR this was? i'm still not seeing any prompt to switch network and can get into this state where the app prompts you to execute txs on the wrong network

steps to repro:
1. connect wallet on staging that's connected to goerli
2. switch network in wallet to another network, ie mainnet
3. attempt to place a bid

expected result:
1. user is prompted to switch to goerli in their wallet

actual result:
1. user is able to execute transaction on the wrong network

**Comment by JacobHomanics** [comment]
@willyogo Should be this one I believe: [PR 56](https://github.com/ATXDAO/nouns-monorepo/pull/56)

Taking a closer look at this

**Comment by ATXDAOEngineering** [comment]
> nice! do you know which PR this was? i'm still not seeing any prompt to switch network and can get into this state where the app prompts you to execute txs on the wrong network
> 
> steps to repro:
> 
> 1. connect wallet on staging that's connected to goerli
> 2. switch network in wallet to another network, ie mainnet
> 3. attempt to place a bid
> 
> expected result:
> 
> 1. user is prompted to switch to goerli in their wallet
> 
> actual result:
> 
> 1. user is able to execute transaction on the wrong network

This is not reproducible on my end. It properly prompts to switch to the proper network and calls the smart contract on the correct chain.

**Comment by JacobHomanics** [comment]
> nice! do you know which PR this was? i'm still not seeing any prompt to switch network and can get into this state where the app prompts you to execute txs on the wrong network
> 
> steps to repro:
> 
> 1. connect wallet on staging that's connected to goerli
> 2. switch network in wallet to another network, ie mainnet
> 3. attempt to place a bid
> 
> expected result:
> 
> 1. user is prompted to switch to goerli in their wallet
> 
> actual result:
> 
> 1. user is able to execute transaction on the wrong network

This is not reproducible on my end. It properly prompts to switch to the proper network and calls the smart contract on the correct chain.

**Comment by willyogo** [comment]
confirmed this is working in staging now! it's working in prod for me now too, but it's prompting me to switch to Optimism instead of mainnet. do we have it set to Optimism for prod somewhere?

![image](https://github.com/ATXDAO/nouns-monorepo/assets/26232238/f08b62c2-8cc6-4496-b48b-1c7a7270448e)
